### PR TITLE
Kyber fixes

### PR DESCRIPTION
**Pull request by Frauschi**
Hi all,

this PR fixes two problems regarding the WolfSSL implementation of Kyber:
* Make sure the source files are included when using CMake
* Include source files in the Zephyr port
* Fix compilation issue when liboqs is also enabled ('`ext_kyber.c` code is disabled)

### CODE REVIEW

**Comment by embhorn** [comment]
@Frauschi is an approved contributor

**Review by dgarske — DISMISSED** [review]

File: cmake/functions.cmake [hunk 1]
```diff
@@ -794,6 +797,11 @@ function(generate_lib_src_list LIB_SOURCES)
               list(APPEND LIB_SOURCES wolfcrypt/src/dilithium.c)
          endif()
 
+         if(BUILD_WC_KYBER)
+              list(APPEND LIB_SOURCES wolfcrypt/src/wc_kyber.c)
+              list(APPEND LIB_SOURCES wolfcrypt/src/wc_kyber_poly.c)
```
Comment (SparkiDev) [inline cmake/functions.cmake:802]: Intel ASM: wc_keyber_asm.S?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: cmake/functions.cmake (hunk 1 above)
Comment (Frauschi) [reply cmake/functions.cmake:802]: fixed

**Review by SparkiDev — APPROVED** [review]

**Comment by Frauschi** [comment]
After some more testing of the Intel ASM addition, I found two other bugs in the CMake `functions.cmake` code: Both `BUILD_INTELASM` and `BUILD_ARMASM` have been misspelled in some places (`BUILD_INTEL_ASM` and `BUILD_ARM_ASM`; I also used the incorrect one in my initial fix yesterday), leading to the two options having no effect in CMake builds. 

I integrated the fixes for that into this PR, hoping that this is ok.

**Review by SparkiDev — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Pull request merged**